hwy/base.h

```diff
@@ -1012,7 +1012,7 @@ HWY_API HWY_BITCASTSCALAR_CONSTEXPR To BitCastScalar(const From& val) {
 // x86 compiler supports _Float16, not necessarily with operators.
 // Avoid clang-cl because it lacks __extendhfsf2.
 #if HWY_ARCH_X86 && defined(__SSE2__) &&                      \
-    ((HWY_COMPILER_CLANG >= 1600 && !HWY_COMPILER_CLANGCL) || \
+    ((HWY_COMPILER_CLANG >= 1500 && !HWY_COMPILER_CLANGCL) || \
```

Comment:
This change seems to cause a regression in libvips' CIFuzz and OSS-Fuzz integration that builds with Clang 15, see:
https://oss-fuzz-build-logs.storage.googleapis.com/log-074523e6-d65b-451f-8868-522bf022989a.txt
<details>
  <summary>Details</summary>

```
In file included from /src/highway/hwy/per_target.cc:27:
In file included from /src/highway/hwy/foreach_target.h:72:
In file included from /src/highway/hwy/per_target.cc:28:
In file included from /src/highway/hwy/highway.h:19:
/src/highway/hwy/base.h:1077:18: error: _Float16 is not supported on this target
  using Native = _Float16;
                 ^
In file included from /src/highway/hwy/timer.cc:23:
/src/highway/hwy/base.h:1077:18: error: _Float16 is not supported on this target
  using Native = _Float16;
                 ^
In file included from /src/highway/hwy/print.cc:16:
In file included from /src/highway/hwy/print.h:24:
/src/highway/hwy/base.h:1077:18: error: _Float16 is not supported on this target
  using Native = _Float16;
                 ^
1 error generated.
make[2]: *** [CMakeFiles/hwy.dir/build.make:118: CMakeFiles/hwy.dir/hwy/print.cc.o] Error 1
make[2]: *** Waiting for unfinished jobs....
In file included from /src/highway/hwy/aligned_allocator.cc:16:
In file included from /src/highway/hwy/aligned_allocator.h:30:
/src/highway/hwy/base.h:1077:18: error: _Float16 is not supported on this target
  using Native = _Float16;
                 ^
In file included from /src/highway/hwy/targets.cc:16:
In file included from /src/highway/hwy/targets.h:28:
/src/highway/hwy/base.h:1077:18: error: _Float16 is not supported on this target
  using Native = _Float16;
                 ^
1 error generated.
make[2]: *** [CMakeFiles/hwy.dir/build.make:76: CMakeFiles/hwy.dir/hwy/aligned_allocator.cc.o] Error 1
In file included from /src/highway/hwy/nanobenchmark.cc:27:
In file included from /src/highway/hwy/robust_statistics.h:24:
/src/highway/hwy/base.h:1077:18: error: _Float16 is not supported on this target
  using Native = _Float16;
                 ^
1 error generated.
```
</details>

However, `_Float16` has been introduced in Clang 15 according to:
https://releases.llvm.org/15.0.0/tools/clang/docs/ReleaseNotes.html#x86-support-in-clang
> Support for the `_Float16` type has been added for all targets with SSE2. When AVX512-FP16 is not available, arithmetic on `_Float16` is emulated using `float`.

So, I'm not sure why this is occurring.

Reply:
Huh, that's strange. I see 

```
#if .. || HWY_SSE2_HAVE_F16_TYPE
#define HWY_HAVE_SCALAR_F16_TYPE 1

#if HWY_ARCH_X86 && defined(__SSE2__) &&                      \
    ((HWY_COMPILER_CLANG >= 1600 && !HWY_COMPILER_CLANGCL) || \
     HWY_COMPILER_GCC_ACTUAL >= 1200)
#define HWY_SSE2_HAVE_F16_TYPE 1
```

So we should not be using it on clang 15, unless we are misdetecting clang? Can you confirm that HWY_COMPILER_CLANG is 1500 in your build environment?

FYI I am at a conference and latency is higher :)

Reply:
With latest HEAD (i.e. 25ad94beabee36d8cdc0a847674de9af6c2ca5fd), I see:
https://github.com/google/highway/blob/25ad94beabee36d8cdc0a847674de9af6c2ca5fd/hwy/base.h#L1014-L1017

So, `HWY_SSE2_HAVE_F16_TYPE` is `1` with Clang 15 or higher.

> So we should not be using it on clang 15, unless we are misdetecting clang? Can you confirm that HWY_COMPILER_CLANG is 1500 in your build environment?

I assumed `HWY_COMPILER_CLANG` is `1500` because of:
```
-- The C compiler identification is Clang 15.0.0
-- The CXX compiler identification is Clang 15.0.0
```

(but I'll verify whether that is correct)

Reply:
I have added a `defined(__FLT16_MAX__)` check in pull request #1976 that should fix the problem with pre-release versions of Clang 15 that do not support `_Float16` on SSE2/SSSE3/SSE4/AVX2.

Both GCC and Clang will define the `__FLT16_MAX__` macro if the `_Float16` type is available.